src/DataTypes/Serializations/SerializationMap.cpp

```diff
@@ -177,11 +177,20 @@ void SerializationMap::serializeText(const IColumn & column, size_t row_num, Wri
 
 void SerializationMap::deserializeText(IColumn & column, ReadBuffer & istr, const FormatSettings & settings) const
 {
+    /// We need read int safe, because in unsafe version
```

Comment:
@alexey-milovidov maybe better to remove unsafe parsing and check performance. Does it really 1.5x faster ?

Reply:
@CurtizJ current solution seems to hacky need to discuss with @alexey-milovidov .

Reply:
@CurtizJ after https://github.com/ClickHouse/ClickHouse/pull/27155 merged, we can keep only tests.